cmd/monaco/dynatrace/dynatrace.go

```diff
@@ -33,92 +30,95 @@ import (
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/environment"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/featureflags"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
-	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log/field"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/trafficlogs"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/account"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/client"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/client/apitoken"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/client/classicheartbeat"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/client/metadata"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/manifest"
 
 	"golang.org/x/oauth2/clientcredentials"
 )
 
-// VerifyEnvironmentGeneration takes a manifestEnvironments map and tries to verify that each environment can be reached
-// using the configured credentials
-func VerifyEnvironmentGeneration(ctx context.Context, envs manifest.EnvironmentDefinitionsByName) bool {
-	if !featureflags.VerifyEnvironmentType.Enabled() {
-		return true
-	}
+type authType = string
+
+const (
+	oAuthType authType = "OAuth"
+	tokenType authType = "token"
+)
+
+var ErrorMissingAuth = errors.New("no token or oAuth credentials provided in the manifest")
+
+// VerifyEnvironmentsAuthentication takes a manifestEnvironments map and tries to verify that each environment can be reached
```

Comment:
```suggestion
// VerifyEnvironmentsAuthentication verifies that all environments can be reached with the defined credentials.
// It returns the first error encountered 
```

I don't think it has too much value to specify too much detail about the types or passing it, but really briefly describe the behavior.